docs/source/en/tei_cloud_run.md

```diff
@@ -0,0 +1,398 @@
+
+-->
+
+# Deploying TEI on Google Cloud Run
+
+Deploying Text Embeddings Inference (TEI) on Google Cloud Platform (GCP) allows to benefit from the underlying [Kubernetes](https://kubernetes.io/) technology which ensures that TEI can scale automatically up or down based on demand.
+
+On Google Cloud, there are 3 main options for deploying TEI (or any other Docker container):
+- Cloud Run
+- Vertex AI endpoints
+- GKE (Google Kubernetes Engine).
+
+This guide explains how to deploy TEI on Cloud Run, a fully managed service by Google. Cloud Run is a so-called serverless offering. This means that the server infrastructure is handled by Google, you only need to provide a Docker container. The benefit of this is that you only pay for compute when there is demand for your application. Cloud Run will automatically spin up servers when there's demand, and scale down to zero when there is no demand.
+
+We will showcase how to deploy any text embedding model with or without a GPU.
+
+> [!NOTE]
+> At the time of writing, GPU support on Cloud Run is generally available in 4 regions. If you're interested in using it, [request a quota increase](https://cloud.google.com/run/quotas#increase) for `Total Nvidia L4 GPU allocation, per project per region`. So far, NVIDIA L4 GPUs (24GiB VRAM) are the only available GPUs on Cloud Run; enabling automatic scaling up to 7 instances by default (more available via quota), as well as scaling down to zero instances when there are no requests.
+
+## Setup / Configuration
+
+This guide already assumes you have set up a Google Cloud project and have enabled billing. This can be done at https://console.cloud.google.com/.
+
+First, you need to install the [gcloud CLI](https://cloud.google.com/sdk/docs/install) on your local machine. This allows to programmatically interact with your Google Cloud project.
+
+Optionally, to ease the usage of the commands within this tutorial, you need to set the following environment variables for GCP:
+
+```bash
+export PROJECT_ID=your-project-id
+export LOCATION=europe-west1  # or any location you prefer: https://cloud.google.com/run/docs/locations
+export CONTAINER_URI="gcr.io/deeplearning-platform-release/huggingface-text-embeddings-inference-cpu.1-6"
+export SERVICE_NAME="text-embedding-server" # choose a name for your service
+export MODEL_ID="ibm-granite/granite-embedding-278m-multilingual" # choose any embedding model
+```
+
+Some clarifications:
+- We provide the latest official Docker image URI based on the [README](https://github.com/huggingface/Google-Cloud-Containers/blob/main/containers/tei/README.md).
+- We choose to deploy the [IBM granite](https://huggingface.co/ibm-granite/granite-embedding-278m-multilingual) embedding model given its strong multilingual capabilities. One can of course choose any other embedding model from the hub. It's recommended to look for models tagged with either `feature-extraction`, `sentence-similarity` or `text-ranking`.
```

Comment:
Do you think we could add a link to the MTEB leaderboard too ?

https://huggingface.co/models?library=sentence-transformers&sort=trending
https://huggingface.co/spaces/mteb/leaderboard